### PR TITLE
putty: update to 0.77

### PR DESCRIPTION
**Pull request by mkhon**
#### Description

Update to 0.77

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.4 21F79 x86_64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: security/putty/Portfile [hunk 1]
```diff
@@ -2,9 +2,10 @@
 
 PortSystem          1.0
 PortGroup           active_variants 1.1
+PortGroup           cmake 1.0
```
Comment (reneeotten) [inline security/putty/Portfile:5]: please use the latest version of the PG, which is `cmake 1.1`

File: security/putty/Portfile [hunk 2]
```diff
@@ -19,28 +20,13 @@ homepage            http://www.chiark.greenend.org.uk/~sgtatham/putty/
...
 
-checksums           rmd160  9357dfca867dc91be09c735b62025f9c614ab46d \
-                    sha256  bb8aa49d6e96c5a8e18a057f3150a1695ed99a24eef699e783651d1f24e7b0be
-
-worksrcdir          ${worksrcdir}/unix
+checksums           rmd160  d4e27e59e3a7235089ccae854db2507a650da02f \
+                    sha256  419a76f45238fd45f2c76b42438993056e74fa78374f136052aaa843085beae5 \
+                    size    2619481
 
 depends_build       port:pkgconfig
```
Comment (reneeotten) [inline security/putty/Portfile:27]: you should use here `depends_build-append` and below as well `depends_lib-append` as not to overwrite any of the dependencies added by the PortGroup. As you can see in the CI now the build fails because it cannot find `cmake`; that build dependency is added by the cmake PG but you effectively remove it here again.

File: security/putty/Portfile [hunk 3]
```diff
@@ -19,28 +20,13 @@ homepage            http://www.chiark.greenend.org.uk/~sgtatham/putty/
...
-
-variant gui description {Build the GUI tools (including putty itself)} {
-    depends_lib-append    path:lib/pkgconfig/gtk+-2.0.pc:gtk2
-    configure.args-replace --without-gtk --with-gtk=2
-    require_active_variants gtk2 x11
-}
-
-default_variants +gui
+depends_lib         path:lib/pkgconfig/gtk+-2.0.pc:gtk2
+require_active_variants gtk2 x11
```
Comment (reneeotten) [inline security/putty/Portfile:29]: why did you remove the variant?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: security/putty/Portfile (hunk 3 above)
Comment (mkhon) [reply security/putty/Portfile]: Done

File: security/putty/Portfile (hunk 2 above)
Comment (mkhon) [reply security/putty/Portfile]: Oh. Fixed

File: security/putty/Portfile (hunk 3 above)
Comment (reneeotten) [reply security/putty/Portfile]: not sure, is there no CMake configure argument to do so?

Comment (mkhon) [reply security/putty/Portfile]: Looking at cmake/gtk.cmake - the port just tries to find GTK by using pkgconfig or gtk-config

Comment (mkhon) [reply security/putty/Portfile]: I found a way to build without GTK

Comment (reneeotten) [reply security/putty/Portfile]: I see it indeed appears there is no configure flag to completely disable it... It was anyway a default variant in the past so unless people specifically disabled the GTK interface they would get it - so I think it's fine. 

The only thing I noticed by typing `ccmake` in the directory is that there is a `PUTTY_GTK_VERSION` flag that by default is set to `ANY` but one can also specify `1`, `2`, and `3`. I think it would be good to explicitly set this to make sure the builds will be reproducible (so set it to `3` if we're going to depend on the `gtk3` port). Since one can set different version for GTK one could add `gtk2` and `gtk3` as variants and then, for example, add `gtk3` as a default variant. But again, I am not sure if that is really of interest to users to be able to select a version.

Comment (mkhon) [reply security/putty/Portfile]: My latest changes include setting PUTTY_GTK_VERSION explicitly

**Comment by mkhon** [comment]
A note why port update is important: 0.70 does not support .ppk version 3

File: security/putty/Portfile [hunk 4]
```diff
@@ -2,9 +2,10 @@
 
 PortSystem          1.0
 PortGroup           active_variants 1.1
+PortGroup           cmake 1.1
 
 name                putty
-version             0.70
+version             0.77
 categories          security net
 platforms           darwin
```
Comment (reneeotten) [inline security/putty/Portfile:10]: this is the default since MacPorts v2.7.0 so the line is no longer needed

File: security/putty/Portfile [hunk 5]
```diff
@@ -19,25 +20,21 @@ homepage            http://www.chiark.greenend.org.uk/~sgtatham/putty/
...
 
-checksums           rmd160  9357dfca867dc91be09c735b62025f9c614ab46d \
-                    sha256  bb8aa49d6e96c5a8e18a057f3150a1695ed99a24eef699e783651d1f24e7b0be
+checksums           rmd160  d4e27e59e3a7235089ccae854db2507a650da02f \
+                    sha256  419a76f45238fd45f2c76b42438993056e74fa78374f136052aaa843085beae5 \
+                    size    2619481
 
-worksrcdir          ${worksrcdir}/unix
+depends_build-append port:pkgconfig
+configure.args-append -DPUTTY_GTK_VERSION=0
```
Comment (reneeotten) [inline security/putty/Portfile:28]: these two lines do not conform to the modeline which states that there should be four spaces; this is aligned at position 22 (which should be 25)

File: security/putty/Portfile [hunk 6]
```diff
@@ -19,25 +20,21 @@ homepage            http://www.chiark.greenend.org.uk/~sgtatham/putty/
...
-# unix configure calls the configure in top folder
-patch.post_args     -d ${worksrcpath}/../
-patchfiles          patch-configure.diff
-
-configure.args-append     --without-gtk
-configure.cppflags-delete -I${prefix}/include
-configure.cppflags-append -isystem${prefix}/include
+pre-configure {
+    configure.cflags-delete -DNDEBUG
+}
```
Comment (reneeotten) [inline security/putty/Portfile:32]: I don't think we want to delete this here... this flag is added by MacPorts and/or CMake

File: security/putty/Portfile (hunk 4 above)
Comment (mkhon) [reply security/putty/Portfile:10]: Do you mean "platform"?

Comment (reneeotten) [reply security/putty/Portfile:10]: Yes, sorry for not being clear ;)

File: security/putty/Portfile [hunk 7]
```diff
@@ -19,25 +20,21 @@ homepage            http://www.chiark.greenend.org.uk/~sgtatham/putty/
 master_sites        ftp://ftp.chiark.greenend.org.uk/users/sgtatham/putty-${version}/ \
                     http://the.earth.li/~sgtatham/putty/${version}/
```
[the remark this answers is not in the document]
Comment (mkhon) [reply security/putty/Portfile]: Done

File: security/putty/Portfile (hunk 4 above)
Comment (mkhon) [reply security/putty/Portfile]: Fixed

File: security/putty/Portfile (hunk 5 above)
Comment (mkhon) [reply security/putty/Portfile]: Ok, re-formatted these two lines

**Review by reneeotten — APPROVED** [review]
